code/modules/mob/living/living.dm

```diff
@@ -438,28 +438,26 @@
 	death()
 
 /**
- * Checks if a mob is incapacitated 
+ * Checks if a mob is incapacitated
  *
  * Normally being restrained, agressively grabbed, or in stasis counts as incapacitated
  * unless there is a flag being used to check if it's ignored
- * 
+ *
  * args:
  * * flags (optional) bitflags that determine if special situations are exempt from being considered incapacitated
- * 
+ *
  * bitflags: (see code/__DEFINES/status_effects.dm)
  * * IGNORE_RESTRAINTS - mob in a restraint (handcuffs) is not considered incapacitated
  * * IGNORE_STASIS - mob in stasis (stasis bed, etc.) is not considered incapacitated
- * * IGNORE_GRAB - mob that is agressively grabbed is not considered incapacitated 
+ * * IGNORE_GRAB - mob that is agressively grabbed is not considered incapacitated
 **/
 /mob/living/incapacitated(flags)
 	if(HAS_TRAIT(src, TRAIT_INCAPACITATED))
 		return TRUE
 
-	if(HAS_TRAIT(src, TRAIT_RESTRAINED) && !(flags & IGNORE_RESTRAINTS))
-		return TRUE
-	if(IS_IN_STASIS(src) && !(flags & IGNORE_STASIS))
+	if(!(flags & IGNORE_RESTRAINTS) && (HAS_TRAIT(src, TRAIT_RESTRAINED) || (!(flags & IGNORE_GRAB) && pulledby && pulledby.grab_state >= GRAB_AGGRESSIVE)))
```

Comment:
```suggestion
	if(!(flags & IGNORE_RESTRAINTS) && HAS_TRAIT(src, TRAIT_RESTRAINED))
		return TRUE
	if(!(flags & IGNORE_GRAB) && pulledby && pulledby.grab_state >= GRAB_AGGRESSIVE)
```

I understand your code works and fixes the bug, but the question of why is not answered.
If is obviously not intended for the ignore-grab argument to need the presence of the ignore-restraints. They are different checks.